### PR TITLE
Adding LAMMPS backend

### PR DESCRIPTION
**Pull request by ndtrung81**
This PR is to add the LAMMPS backend for PySAGES. LAMMPS needs to be built as a python module and with a GPU-enabled accelerator package. For the moment, only the KOKKOS package supports access to per-atom arrays on the device.

### CODE REVIEW

**Comment by pabloferz** [comment]
I created https://github.com/SSAGESLabs/lammps-dlext for setting up the DLExt plugin there.

**Comment by ndtrung81** [comment]
@pabloferz I happened to create my own repo: https://github.com/ndtrung81/lammps-dlext

**Comment by pabloferz** [comment]
I see, do you mind if we transfer it to `SSAGESLabs`?

**Comment by ndtrung81** [comment]
@pabloferz Let me first get it to a functional state with the LAMMPS backend PR, and then create PRs to the one under PySAGES.

**Comment by thangckt** [comment]
I am looking forward to seeing this

File: examples/lammps/unbiased/unbiased.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    N = view.global_particle_number()
+    print(f"Total number of particles = {N}")
+    cuda_enabled = view.has_kokkos_cuda_enabled()
+    print(f"CUDA enabled = {cuda_enabled}")
+
+    # create a Sampler object
+    args = "mydlext all dlext"
+    args=args.split()
+    # TODO: this still crashes
+    #sampler = lammps_dlext.FixDLExt(context, args)
```
Comment (pabloferz) [inline examples/lammps/unbiased/unbiased.py:72]:
```suggestion
    args = "1 all dlext".split()
    sampler = lammps_dlext.FixDLExt(context, args)
```

File: examples/lammps/unbiased/unbiased.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    # attempt to get the atom positions
+    #x = sampler.positions()
```
Comment (pabloferz) [inline examples/lammps/unbiased/unbiased.py:75]:
```suggestion
    x = lammps_dlext.positions(view, lammps_dlext.kOnHost)
```

**Comment by pabloferz** [comment]
Closing in favor of #274.

**Pull request closed without merging**